### PR TITLE
[Navigation, Topbar] New DL updates

### PR DESCRIPTION
**Pull request by kyledurand**
### WHY are these changes introduced?

Fixes https://github.com/Shopify/polaris-react/issues/3190 and fixes https://github.com/Shopify/polaris-react/issues/3188 and fixes https://github.com/Shopify/polaris-react/issues/3189 and fixes https://github.com/Shopify/polaris-ux/issues/405

- [View the changes](https://5d559397bae39100201eedc1-ddklsrszjr.chromatic.com/iframe.html?id=playground-playground--details-page&contexts=New%20Design%20Language%3DEnabled%20-%20Light%20Mode) 
- [Compare with current changes](http://polaris-react.herokuapp.com/iframe.html?id=playground-playground--details-page&contexts=New%20Design%20Language=Enabled%20-%20Light%20Mode)

image

## 

### How to 🎩

🖥 [Local development instructions](https://github.com/Shopify/polaris-react/blob/master/README.md#development)
🗒 [General tophatting guidelines](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting.md)
📄 [Changelog guidelines](https://github.com/Shopify/polaris-react/blob/master/documentation/Versioning%20and%20changelog.md)

http://localhost:6006/iframe.html?id=playground-playground--details-page&contexts=New%20Design%20Language%3DEnabled%20-%20Light%20Mode&viewMode=story

### 🎩 checklist

* [ ] Tested on [mobile](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting.md#cross-browser-testing)
* [ ] Tested on [multiple browsers](https://help.shopify.com/en/manual/intro-to-shopify/shopify-admin/supported-browsers)
* [ ] Tested for [accessibility](https://github.com/Shopify/polaris-react/blob/master/documentation/Accessibility%20testing.md)
* [ ] Updated the component's `README.md` with documentation changes
* [ ] [Tophatted documentation](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting%20documentation.md) changes in the style guide
* [ ] For visual design changes, pinged one of @ HYPD, @ mirualves, @ sarahill, or @ ry5n to update the Polaris UI kit

### CODE REVIEW

File: documentation/Color system.md [hunk 1]
```diff
@@ -274,3 +274,4 @@ Used to decorate elements where color does convey a specific meaning in componen
 | `--p-range-slider-thumb-size-active`      | `2.4rem`                                                                                                     |
 | `--p-range-slider-thumb-scale`            | `1.5`                                                                                                        |
 | `--p-badge-font-weight`                   | `500`                                                                                                        |
+| `--p-frame-background`                    | `rgb(244, 245, 246)`                                                                                         |
```
Comment (kyledurand) [inline documentation/Color system.md:277]: This is temporary until we can make the background change in tokens: https://github.com/Shopify/polaris-tokens/blob/master/src/configs/base.ts

File: src/components/Frame/Frame.scss [hunk 2]
```diff
@@ -7,7 +7,7 @@ $skip-vertical-offset: rem(10px);
   width: 100%;
   min-height: 100vh;
   display: flex;
-  background-color: var(--p-background, color('sky', 'light'));
+  background-color: var(--p-frame-background, color('sky', 'light'));
```
Comment (kyledurand) [inline src/components/Frame/Frame.scss:10]: This breaks dark mode 😞

**Comment by yesenia-perezcruz** [comment]
Is the background color on the nav white? It should be white but I can't tell from the screenshot if it's white or light gray.

**Comment by kyledurand** [comment]
@yesenia-perezcruz can you link to that figma file? [This one](https://www.figma.com/file/DOLJoC3CbldApU26mIIDiK/Page-Component?node-id=61%3A2838) shows the nav with the same background color as the content

**Comment by yesenia-perezcruz** [comment]
@kyledurand I'm looking at the same Figma file, in the updates page. Here's a screenshot: 

Screen Shot 2020-08-25 at 5 03 52 PM

**Comment by alex-page** [comment]
@kyledurand Looking at Percy there are some changes in purpler that we probably didn't intend to be there. I am loading this up locally now and going through the code.

File: documentation/Color system.md (hunk 1 above)
Comment (alex-page) [reply documentation/Color system.md:277]: Should we make a PR in tokens to kick off this change?

**Comment by kyledurand** [comment]
Thanks @alex-page I saw those as well but didn't have time to fix them yesterday. Making this my priority this morning

File: documentation/Color system.md (hunk 1 above)
Comment (kyledurand) [reply documentation/Color system.md:277]: This new color is very similar to the old background color but it's slightly darker. I'll look into adding it to tokens but I'm not sure how to reverse engineer a hex color into hue, saturation, lightness values. Might take some digging

Comment (kyledurand) [reply documentation/Color system.md:277]: I forgot chrome gives you hsl values so I should be able to base it off: `hsl(210deg, 10%, 96%)`. I'll put up a pr shortly

**Comment by kyledurand** [comment]
Percy is 🍏. Ready for another review / tophat on this [Chroma link](https://5d559397bae39100201eedc1-gbibwvpcth.chromatic.com/)  🙏

File: documentation/Color system.md (hunk 1 above)
Comment (alex-page) [reply documentation/Color system.md:276]: Yeah you can I think `ctrl` click the color and it will convert between the different values 💯

**Comment by kyledurand** [comment]
Good catch @alex-page! I've updated [this line](https://github.com/Shopify/polaris-react/pull/3198/commits/df9a64dc9dd4bb3de9d1069c88f867a04d2392af) to address the second issue.

https://screenshot.click/screencast_2020-08-27_07-31-56.mp4

The first one, as you suspected, is caused by some storybook config which adds 1rem of padding to the `body` tag. It's always been there, and a bit of a pain for me while making frame changes 😃

**Pull request merged**